### PR TITLE
Add checkbox component

### PR DESCRIPTION
**Pull request by mjlescano**
Closes https://github.com/Synthetixio/gmv3-interface/issues/35

### CODE REVIEW

File: .storybook/preview-body.html [hunk 1]
```diff
@@ -12,6 +12,10 @@
 		font-size: 12px;
 	}
 
+	body {
+		min-height: 100vh;
+	}
```
Comment (mjlescano) [inline .storybook/preview-body.html:17]: This fixes the custom background when changing it. It wasn't covering the full height.

File: .storybook/preview.js [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+	backgrounds: {
+		default: 'dark',
+		values: [
+			{ name: 'dark', value: '#111' },
+			{ name: 'light', value: '#F8F8F8' },
+			{ name: 'black', value: '#000000' },
+			{ name: 'white', value: '#FFFFFF' },
+			{ name: 'purple', value: colors.gradients.purple },
+			{ name: 'darkBlue', value: colors.gradients.darkBlue },
+		],
```
Comment (mjlescano) [inline .storybook/preview.js:20]: Added some common backgrounds on the toolbar to test our components. It includes common colors from our designs.

File: src/styles/colors.ts [hunk 3]
```diff
@@ -10,13 +10,13 @@ export default {
...
-		pink: 'linear-gradient(73.6deg, #8E2DE2 2.11%, #ED1EFF 90.45%);',
-		rainbow: 'linear-gradient(90deg, #ED1EFF 0%, #00D1FF 100%);',
-		grey: 'linear-gradient(0deg, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2)), linear-gradient(311.52deg, #3D464C -36.37%, #131619 62.81%);',
+		darkBlue: 'linear-gradient(180deg, #08021E 0%, #1F0777 146.21%)',
+		lightBlue: 'linear-gradient(73.6deg, #85FFC4 2.11%, #5CC6FF 90.45%)',
+		purple: 'linear-gradient(73.6deg, #8E2DE2 2.11%, #4B01E0 90.45%)',
+		orange: 'linear-gradient(73.6deg, #FF8060 2.11%, #FEB27A 90.45%)',
+		pink: 'linear-gradient(73.6deg, #8E2DE2 2.11%, #ED1EFF 90.45%)',
+		rainbow: 'linear-gradient(90deg, #ED1EFF 0%, #00D1FF 100%)',
+		grey: 'linear-gradient(0deg, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2)), linear-gradient(311.52deg, #3D464C -36.37%, #131619 62.81%)',
```
Comment (mjlescano) [inline src/styles/colors.ts:19]: Removed the `;`'s from here and updated its usage everywhere to be more consistent with the other colors (and this allows gradients compositions!)

File: src/components/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	onChange: (checked: boolean) => unknown;
+}
+
+export default function Checkbox({
+	id,
+	disabled = false,
+	label,
+	name,
+	checked,
+	onChange,
```
Comment (fritzschoff) [inline src/components/Checkbox.tsx:21]: can you add `...rest` and add that to the`StyledContainer`, so if the consumer component plugs that into styled component again cause he needs to fine tune this component, the styles are getting applied

File: src/components/Checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	display: flex;
+`;
+
+const StyledLabelText = styled.span<{ disabled?: boolean }>`
+	display: inline-block;
+	cursor: pointer;
+	font-family: 'Inter Bold';
+	font-size: 1.2rem;
+	line-height: 2.2rem;
+	${colors.white};
```
Comment (fritzschoff) [inline src/components/Checkbox.tsx:67]: guess you want to do color: ${colors.white} ?

File: src/components/Checkbox.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+		content: '';
+		position: absolute;
+		left: 27%;
+		top: 27%;
+		border-radius: 50%;
+		width: 46%;
+		height: 46%;
+		background-image: ${colors.gradients.pink};
+		opacity: ${(attrs) => (attrs.checked ? '1' : '0')};
+		transition: opacity 80ms ease-in-out;
```
Comment (fritzschoff) [inline src/components/Checkbox.tsx:105]: nice! animation. GJ

File: src/styles/colors.ts (hunk 3 above)
Comment (fritzschoff) [reply src/styles/colors.ts:19]: thanks!

**Review by fritzschoff — APPROVED** [review]

**Pull request merged**